src/extension.ts

```diff
@@ -356,6 +358,10 @@ function launchMetals(
       }
     });
 
+    window.onDidChangeWindowState(windowState => {
```

Comment:
Should we guard this against the server version number? This would require merging the PR to the main Metals repo first and wait until there is a SNAPSHOT release out. We already depend on a semver library.

Reply:
What happens if you send the notification to the current server? Doesn't it just discard it or log it with a warning?

Reply:
It logs a warning, which would probably confuse users.

Reply:
I'm 👍 for adding a guard here, but I would remove it in the future when the server version targeted by the VSCode extension supports it.

Reply:
Alternatively, the server could register its interest in these notifications similar to how file watchers work. That might be overkill however

Reply:
I agree it's overkill. Let's add a guard and remove it when we release the next version of the server.

Reply:
Good spot, but how can I do this? And also correct me if I'm wrong but I think that when we merge the server part that guard will be unnecessary, am I right?

Reply:
@agajek correct, they idea is to merge this with the guard on, and to remove it when Metals 0.6.0 is released. I would use the semver package to do something like:

```ts
const serverVersion = config.get<string>("serverVersion")!;
if (semver.gte(serverVersion, "0.6.0")) {
  // ...
}
```

(I wrote this directly as a comment, it might be wrong, but it's just to give an idea)

Reply:
The `serverVersion` variable is already computed here https://github.com/scalameta/metals-vscode/blob/1cc6b239ac7505c35900b65717a0b864cc2fd0aa/src/extension.ts#L94